matchesWhereQuery.js

```diff
@@ -3,45 +3,73 @@ var _ = require( 'underscore' );
 module.exports = function( object, whereQuery ) {
 	var whereQueryKeys = _.keys( whereQuery );
 	for( var i = 0; i < whereQueryKeys.length; i++ ) {
-		var thisKey = whereQueryKeys[ i ];
-		var thisKeyQuery = whereQuery[ thisKey ];
-		var thisObjectValue = object[ thisKey ];
+		var attribute = whereQueryKeys[ i ];
+		var queryAttribute = whereQuery[ attribute ];
+		var objectAttribute = object[ attribute ];
 
-		if( _.isObject( thisKeyQuery ) && thisKeyQuery.comparator ) {
-			if( ! thisKeyQuery.value ) throw new Error( 'Value must be supplied for comparator queries' );
+		if( _.isObject( queryAttribute ) && queryAttribute.comparator ) {
+			if( ! queryAttribute.value ) throw new Error( 'Value must be supplied for comparator queries' );
 
-			switch( thisKeyQuery.comparator ) {
+			switch( queryAttribute.comparator ) {
 			case 'doesNotEqual':
-				if( thisObjectValue === thisKeyQuery.value ) return false;
+				if( objectAttribute === queryAttribute.value ) return false;
 				break;
 			case 'isGreaterThan':
-				if( thisObjectValue <= thisKeyQuery.value ) return false;
+				if( objectAttribute <= queryAttribute.value ) return false;
 				break;
 			case 'isGreaterThanOrEqualTo':
-				if( thisObjectValue < thisKeyQuery.value ) return false;
+				if( objectAttribute < queryAttribute.value ) return false;
 				break;
 			case 'isLessThan':
-				if( thisObjectValue >= thisKeyQuery.value ) return false;
+				if( objectAttribute >= queryAttribute.value ) return false;
 				break;
 			case 'isLessThanOrEqualTo':
-				if( thisObjectValue > thisKeyQuery.value ) return false;
+				if( objectAttribute > queryAttribute.value ) return false;
 				break;
 			case 'isBetween':
-				if( ! _.isArray( thisKeyQuery.value ) || thisKeyQuery.value.length !== 2 ) {
+				if( ! _.isArray( queryAttribute.value ) || queryAttribute.value.length !== 2 ) {
 					throw new Error( 'Value supplied for isBetween comparator must be an array [ min, max ]' );
 				}
 
-				if( thisObjectValue < thisKeyQuery.value[ 0 ] || thisObjectValue > thisKeyQuery.value[ 1 ] ) return false;
+				if( objectAttribute < queryAttribute.value[ 0 ] || objectAttribute > queryAttribute.value[ 1 ] ) return false;
+				break;
+			case 'startsWith':
+				// eslint-disable-next-line max-len
+				if( ! _.isString( objectAttribute ) || ! _.isString( queryAttribute.value ) ) throw new Error( 'For startsWith comparator both the object attribute and the query value must be strings.' );
```

Comment:
i think we might want to consider the `null` case here (as well as with `endsWith`). if the `objectAttribute` is `null` that should be a non-match

Reply:
Good one, its done, and tested! thanks